File: Problems/Algorithms/121. Best Time to Buy and Sell Stock/buy_sell_stock.py
#!/usr/bin/env python3ss

from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit, buy = 0, prices[0]
        
        for price in prices:
            profit = max(profit, price-buy)
            buy = min(buy, price)
        
        return profit

File: Problems/Algorithms/121. Best Time to Buy and Sell Stock/BuySellStock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0;
        int buy = prices[0];
        
        for (int price : prices) {
            profit = Math.max(profit, price-buy);
            buy = Math.min(buy, price);
        }
        
        return profit;
    }
}
